jobs/ror/DownloadRoRDatasetInSync.php

```diff
@@ -0,0 +1,159 @@
+                    : '';
+                
+                if (!empty($pathCsv) && $this->fileManager->fileExists($pathCsv)) {
+                    UpdateRorRegistryDataset::writeToExecutionLogFile(
+                        'Cancelling sync download as the CSV already exists',
+                        $this->scheduledTaskLogFilePath, 
+                        ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_NOTICE
+                    );
+                    return true; // No need to download the CSV file, job can skipped
+                }
+
+                UpdateRorRegistryDataset::writeToExecutionLogFile(
+                    'Proceeding with sync download as the CSV file not found',
+                    $this->scheduledTaskLogFilePath, 
+                    ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_ERROR
+                );
+                
+                return false; // Need to download the CSV file
+            }),
+            (new WithoutOverlapping($this->pathZipFile . ':sync'))->expireAfter(600), // 10-minute lock
+        ];
+    }
+
+    public function handle()
+    {
+        try {
+
+            // if the chunk process still ongoing,
+            // we will release back the job in the queue for next attempt
+            if (Cache::has(UpdateRorRegistryDataset::CACHE_KEY_CHUNK_BATCH) 
+                && $this->attempts() <= $this->tries
+            ) {
+                UpdateRorRegistryDataset::writeToExecutionLogFile(
+                    "Chunk batch not complete, redispatching DownloadRoRDatasetInSync with attempt {$this->attempts()}",
+                    $this->scheduledTaskLogFilePath,
+                    ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_NOTICE
+                );
+                $this->release($this->backoff);
+            }
+
+            UpdateRorRegistryDataset::cleanup([$this->pathZipFile, $this->pathZipDir]);
+
+            UpdateRorRegistryDataset::writeToExecutionLogFile(
+                'Downloading ROR dataset synchronously',
+                $this->scheduledTaskLogFilePath,
+                ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_NOTICE
+            );
+
+            $client = Application::get()->getHttpClient();
+            $response = $client->request('GET', $this->downloadUrl, [
+                'sink' => fopen($this->pathZipFile, 'wb'),
+                'connect_timeout' => 10,
+                'timeout' => $this->timeout - 10, // set the download time 10 second less than the job timeout
+            ]);
+
+            if ($response->getStatusCode() !== 200
+                || !$this->fileManager->fileExists($this->pathZipFile)
+            ) {
+                throw new Exception('Failed to download ROR dataset');
+            }
+
+            $zip = new ZipArchive();
+            if ($zip->open($this->pathZipFile) !== true) {
+                throw new Exception('Failed to open ZIP file');
+            }
+            $zip->extractTo($this->pathZipDir);
```

Comment:
We can extract only the needed file instead. It's also needed to check how the extraction is done.

If the `ZipArchive` creates the file already at the right path, then it's better to extract it on a temporary folder first... I mean, if the extraction fails at the middle, the code will consider it was successfully (there's no checksum, just a check for the file existence).